contrib/platform/src/com/sun/jna/platform/unix/X11.java

```diff
@@ -291,8 +302,9 @@ void XShapeCombineMask(Display display, Window window, int dest_kind,
             public short green, greenMask;
             public short blue, blueMask;
             public short alpha, alphaMask;
+            @Override
             protected List getFieldOrder() {
-                return Arrays.asList(new String[] { "red", "redMask", "green", "greenMask", "blue", "blueMask", "alpha", "alphaMask" }); 
+                return Arrays.asList(new String[] { "red", "redMask", "green", "greenMask", "blue", "blueMask", "alpha", "alphaMask" });
```

Comment:
Fair enough.  

> On Dec 7, 2015, at 12:34 AM, Lyor Goldstein <notifications@github.com> wrote:
> 
> In contrib/platform/src/com/sun/jna/platform/unix/X11.java:
> 
> >              protected List getFieldOrder() {
> > -                return Arrays.asList(new String[] { "red", "redMask", "green", "greenMask", "blue", "blueMask", "alpha", "alphaMask" }); 
> > +                return Arrays.asList(new String[] { "red", "redMask", "green", "greenMask", "blue", "blueMask", "alpha", "alphaMask" });
> 
> Agreed, but I did not write this code.  Furthermore, if we really want to make a worthy change it should be along the following lines:
> 
> public class MyStructure extends Structure
>  {
>      
> public
>  field1;
>      
> public
>  field2;
>      
> ... etc...
> 
> 
>      
> // No harm in it being 'public' since it is un-modifiable
> 
>      
> public static final List<String> FIELDS = Collections.unmodifiableList(Arrays.asList("field1", "field2", ...etc...
> ));
> 
>      
> @Override
> 
>       
> protected List<String> getFieldOrder
> () {     
>             
> return FIELDS
> ;
>       }
> }
> 
> The rationale is that the field order is invariant, so there is not need to create a new List every time getFieldOrder is called
> 
> —
> Reply to this email directly or view it on GitHub.
>